nixos/modules/services/admin/salt/minion.nix

```diff
@@ -28,24 +27,33 @@ in
         default = {};
         description = ''
           Salt minion configuration as Nix attribute set.
-          See <link xlink:href="https://docs.saltstack.com/en/latest/ref/configuration/minion.html"/>                                                                                                 
-          for details.          
+          See <link xlink:href="https://docs.saltstack.com/en/latest/ref/configuration/minion.html"/>
+          for details.
         '';
       };
     };
   };
 
   config = mkIf cfg.enable {
-    environment.systemPackages = with pkgs; [ salt ];
+    environment = {
+      # Set this up in /etc/salt/minion so `salt-call`, etc. work.
+      # The alternatives are
+      # - passing --config-dir to all salt commands, not just the minion unit
+      # - setting a global environment variable
+      etc."salt/minion".source = pkgs.writeText "minion" (
+        builtins.toJSON fullConfig
+      );
+      systemPackages = with pkgs; [ salt ];
+    };
     systemd.services.salt-minion = {
       description = "Salt Minion";
       wantedBy = [ "multi-user.target" ];
       after = [ "network.target" ];
-      path = with pkgs; [
-        utillinux
-      ];
+      path = [
+        (builtins.dirOf config.security.wrapperDir)
+      ] ++ config.environment.profiles;
```

Comment:
I haven't seen `environment.profiles` being used like this, but it seems like a good solution

Reply:
One thing:
You might want to consider reversing the list. The way this list seems to work is that the most specific thing comes last. Thats why reversing it when creating a `PATH` might be a good idea. 

See: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/environment.nix#L58